target/linux/ath79/image/generic.mk

```diff
@@ -607,6 +619,25 @@ define Device/engenius_ecb1750
 endef
 TARGET_DEVICES += engenius_ecb1750
 
+define Device/engenius_ens202ext
+  SOC := ar9341
+  DEVICE_VENDOR := EnGenius
+  DEVICE_MODEL := ENS202EXT
+  DEVICE_PACKAGES := rssileds kmod-leds-gpio
+  IMAGE_SIZE := 12032k
+  KERNEL := kernel-bin | append-dtb | lzma | uImage lzma -M 0x4f4b4c49
+  LOADER_TYPE := bin
+  LOADER_FLASH_OFFS := 0x230000
+  COMPILE := loader-$(1).bin loader-$(1).uImage
+  COMPILE/loader-$(1).bin := loader-okli-compile
+  COMPILE/loader-$(1).uImage := append-loader-okli $(1) | \
+	pad-to $$$$(BLOCKSIZE) | lzma | uImage lzma
+  IMAGES += factory.bin
+  IMAGE/factory.bin := append-squashfs-fakeroot-be | pad-to $$$$(BLOCKSIZE) | \
+	$$(IMAGE/sysupgrade.bin) | engenius_ens202ext-factory $(1) | gzip
```

Comment:
Why did you extracted `gzip` from the recipe `Build/engenius_ens202ext-factory` and compress it in a separate step?
It feels better to me that the `....-factory` recipe produces an image which is acceptable by the OEM firmware.

Reply:
Please don't use IMAGE/sysupgrade.bin here, as that will include the metadata, which you don't need for factory. Instead, just copy the relevant commands from IMAGE/sysupgrade.bin ...

Reply:
for gzip:

I had created a bunch of image commands so that one doesn't have to write a custom one (like Build/engenius_ens202ext-factory)
The gzip command that exists already in `include/image-commands.mk` does the same job without me having to rewrite the command in the custom recipe.

I guess its a trade-off, do we prefer the code to be shorter/common or more understandable/grouped?

for $$(IMAGE/sysupgrade.bin):

also took that from #2299